nexus/src/app/sagas/instance_update/mod.rs

```diff
@@ -614,17 +616,43 @@ impl UpdatesRequired {
                     new_runtime.dst_propolis_id = None;
                     update_required = true;
                 }
-
-                // Even if the active VMM was destroyed (and we set the
-                // instance's state to `NoVmm` above), it has successfully
-                // migrated, so leave it in the VMM state and don't deallocate
-                // virtual provisioning records --- the instance is still
-                // incarnated.
-                new_runtime.nexus_state = InstanceState::Vmm;
-                deprovision = false;
             }
         }
 
+        // If the *new* state no longer has a `propolis_id` field, that means
+        // that the active VMM was destroyed without a successful migration out
+        // (or, we migrated out to a target VMM that was immediately destroyed,
```

Comment:
Nit: it may be good to clarify when this *does* happen; the gist of it is that a running VMM can stop at any time even without the control plane being involved, so it's possible to have a migration succeed and then have the target immediately stop, all before the sled agents involved get a chance to read any state or relay it up to Nexus.